lib/bolt/inventory/group2.rb

```diff
@@ -1,19 +1,20 @@
 # frozen_string_literal: true
 
 require 'bolt/inventory/group'
+require 'bolt/inventory/inventory2'
 
 module Bolt
   class Inventory
     class Group2
       attr_accessor :name, :targets, :aliases, :name_or_alias, :groups,
-                    :config, :rest, :facts, :vars, :features, :plugin_hooks
+                    :config, :facts, :vars, :features, :plugin_hooks
 
       # THESE are duplicates with the old groups for now.
       # Regex used to validate group names and target aliases.
       NAME_REGEX = /\A[a-z0-9_][a-z0-9_-]*\Z/.freeze
 
       DATA_KEYS = %w[name config facts vars features plugin_hooks].freeze
-      NODE_KEYS = DATA_KEYS + %w[alias uri]
+      TARGET_KEYS = DATA_KEYS + %w[alias uri safe_name]
```

Comment:
I don't think we want people to set this. The only reason it exists is when the URI isn't safe right and there is no explicit name?

Reply:
You are right. I was using this because we assign a name from URI when adding targets to groups (which was complicating my algorithm for when a safe name should not be the name), I figured if a user added `safe_name` in inventory it would just silently overwritten.  I see how that would be confusing to to the user. I will figure out a different way.